.github/workflows/main.yml

```diff
@@ -0,0 +1,145 @@
+name: Pull Request Tests
+
+on:
+  push:
+    branches:
+    - feature/ContInteg
+  pull_request:
+    branches:
+    - feature/ContInteg
+
+jobs:
+  setup:
+    name: Set up
+    runs-on: ubuntu-latest
+
+    outputs:
+      tn: ${{ steps.parse.outputs.tn }}
+      bld: ${{ steps.parse.outputs.bld }}
+      test: ${{ steps.parse.outputs.test }}
+      img: ${{ steps.parse.outputs.img }}
+
+    steps:
+    - name: Checkout codes
+      uses: actions/checkout@v2
+
+    - name: Parse cases
+      id: parse
+      run: |
+        name_=$(sed -n 1p ${GITHUB_WORKSPACE}/tests/ci/ci.test)
+        case_=$(sed -n 2p ${GITHUB_WORKSPACE}/tests/ci/ci.test)
+        img_=$(sed -n 3p ${GITHUB_WORKSPACE}/tests/ci/ci.test)
+
+        imd1_=''
+        [[ $case_ =~ thr || $case_ =~ mpi || $case_ =~ dcp || $case_ =~ rst ]] && imd1_+='std'
+        [[ $case_ =~ bit ]] && imd1_+=' bit'
+        [[ $case_ =~ dbg ]] && imd1_+=' dbg'
+        imd1_=$(echo $imd1_ | sed -e 's/^ *//' -e 's/ *$//')
+
+        bld_='{"bld_set":['
```

Comment:
This looks like you are constructing a json object, in a shell script based on option parsed from ct.test. Is this what's going on here?

Reply:
Yes, that is exactly right.
- In ci.test, a user specifies which unit tests to run: thr, mpi, dcp, rst, dbg, bit
- Based on this, json objects are constructed which includes information on which cases to build and run
- The json objects are used in the subsequent steps below to run actions in a matrix

Reply:
Too complicated. Like I said in my previous comment I do not think github action should contain any logic that depends on specification of ci.test file.

Reply:
I agree with @DusanJovic-NOAA. There is too much `scripting` in these actions.

Reply:
OK. Let me try putting these in a separate script.